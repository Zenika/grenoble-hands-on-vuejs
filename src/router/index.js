import { createRouter, createWebHistory } from 'vue-router'
import Home from '@/views/Home.vue'
import City from '@/views/City.vue'

const routes = [
  {
    path: '/',
    name: 'Home',
    component: Home
  },
  {
    path: '/:cityName',
    name: 'City',
    props: true,
    component: City
  }
]

const router = createRouter({
  history: createWebHistory(process.env.BASE_URL),
  routes
})

export default router
