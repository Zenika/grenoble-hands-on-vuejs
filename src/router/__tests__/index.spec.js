import {routes} from "@/router";

jest.mock('@/views/Home.vue', () => 'HomeMockComponent')
jest.mock('@/views/City.vue', () => 'CityMockComponent')
jest.mock('@/views/CreateCity.vue', () => 'CreateCityMockComponent')

describe('Router', () => {
  it('should have 2 routes', () => {
    expect(routes.length).toBe(3)
  });
  it('should match to snapshot', () => {
    expect(routes).toMatchSnapshot()
  });
});
