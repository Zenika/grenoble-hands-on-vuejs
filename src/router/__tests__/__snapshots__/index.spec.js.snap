// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router should match to snapshot 1`] = `
Array [
  Object {
    "component": "HomeMockComponent",
    "name": "Home",
    "path": "/",
  },
  Object {
    "component": "CreateCityMockComponent",
    "name": "CreateCity",
    "path": "/city/create",
  },
  Object {
    "component": "CityMockComponent",
    "name": "City",
    "path": "/:cityName",
    "props": true,
  },
]
`;
